vhost-device-vsock/Cargo.toml

```diff
@@ -27,6 +27,8 @@ virtio-vsock = "0.6"
 vm-memory = "0.14.1"
 vmm-sys-util = "0.12"
 figment = { version = "0.10.19", features = ["yaml"] }
+vsock = "0.5.0"
```

Comment:
I need some input on this. Did you mean to include a feature like 'vsock_proxy' and then put the new dependencies and all the new vsock backend code under the feature check like '#cfg(features = 'vsock_proxy')'? It should be easy to do so for the enum codes but the code in main.rs is a bit problematic to put under such conditional checks. Do you have any suggestion on this? Also do I need to put tests related to vsock backend under the feature check too? And is there any CI changes needed so that we make sure that the new tests are run too as we need to include a feature check?

Reply:
It would be a bit hard to put all the new vsock backend code so I didn't address this.

Reply:
ack!

Reply:
On second thought, vhost-user-vsock is very useful in those systems where AF_VSOCK is not supported by the host (e.g. FreeBSD, etc.), because it uses the hybrid vsock, requiring only AF_UNIX in the host. Right now it's true that we only support linux, but in the future that's supposed to change, so this new feature we're adding, it would be nice to be able to disable it in those systems.

That said, I don't want to force you to do that in this PR, so if you don't want/can't, we can postpone it to a future PR and create an issue.

Reply:
No reason to postpone it. Done now.